### PR TITLE
fix(api): Add owner field to selection set if implicitly defined

### PR DESCRIPTION
**Pull request by richardmcclellan**
This PR fixes a bug where subscriptions will not be received at all for the schema below.    The fix is to modify the request for mutations to include the owner field to the selection set.  This ensures that when the mutation is broadcast out to the subscription listeners, the owner field will be hydrated, so it won't be filtered out, because it will be able to validate that the user is authorized to view this object.

```
type Post @model @auth(rules: [{ allow: owner }])
{
  id: ID!
  title: String!
}
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by changxu0306 — APPROVED** [review]
Looks good!

File: aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java [hunk 1]
```diff
@@ -244,6 +246,12 @@ private SelectionSet wrapPagination(SelectionSet node) {
                 } else {
                     result.add(new SelectionSet(fieldName, null));
                 }
+                for (AuthRule authRule : schema.getAuthRules()) {
+                    if (AuthStrategy.OWNER.equals(authRule.getAuthStrategy())) {
+                        result.add(new SelectionSet(authRule.getOwnerFieldOrDefault(), null));
```
Comment (jamesonwilliams) [inline aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java:251]:
```suggestion
                        result.add(new SelectionSet(authRule.getOwnerFieldOrDefault(), Collections.emptySet()));
```

**Review by jamesonwilliams — APPROVED** [review]

File: aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java (hunk 1 above)
Comment (richardmcclellan) [reply aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java]: I just pushed a change to require `SelectionSet.nodes` to be nonnull, let me know what you think!

Comment (jamesonwilliams) [reply aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java]: 'Tis a thing of exceptional beauty, @richardmcclellan, thanks!

**Pull request merged**